application/src/main/java/org/togetherjava/tjbot/commands/free/FreeCommand.java

```diff
@@ -0,0 +1,330 @@
+     */
+    public FreeCommand() {
+        super(FREE_COMMAND, "marks this channel as free for another user to ask a question",
+                SlashCommandVisibility.GUILD);
+
+        channelToStatusMessage = new HashMap<>();
+        channelMonitor = new ChannelMonitor();
+
+        isReady = false;
+    }
+
+    /**
+     * Reaction to the 'onReady' event. This method binds the configurables to the
+     * {@link net.dv8tion.jda.api.JDA} instance. Including fetching the names of the channels this
+     * command monitors.
+     * <p>
+     * It also updates the Status messages in their relevant channels, so that the message is
+     * up-to-date.
+     * <p>
+     * This also registers a new listener on the {@link net.dv8tion.jda.api.JDA}, this should be
+     * removed when the code base supports additional functionality
+     *
+     * @param event the event this method reacts to
+     */
+    @Override
+    public void onReady(@NotNull final ReadyEvent event) {
+        final JDA jda = event.getJDA();
+        // todo remove this when onGuildMessageRecieved has another access point
+        jda.addEventListener(this);
+
+        initChannelsToMonitor();
+        initStatusMessageChannels(jda);
+        logger.debug("Config loaded:\n{}", channelMonitor);
+
+        checkBusyStatusAllChannels(jda);
+        initStatusMessages(jda);
+
+        channelMonitor.statusIds()
+            .map(event.getJDA()::getTextChannelById)
+            .filter(Objects::nonNull)
+            .forEach(this::displayStatus);
+
+        isReady = true;
+    }
+
+    /**
+     * When triggered with {@code /free} this will mark a help channel as not busy (free for another
+     * person to use).
+     * <p>
+     * If this is called on from a channel that was not configured for monitoring (see
+     * {@link FreeCommandConfig}) the user will receive an ephemeral message stating such.
+     * 
+     * @param event the event that triggered this
+     */
+    @Override
+    public void onSlashCommand(@NotNull final SlashCommandEvent event) {
+        logger.debug("/free used by {} on channel {}", event.getUser().getAsTag(),
+                event.getChannel().getName());
+        if (!shouldHandle(event)) {
+            return;
+        }
+
+        long id = event.getChannel().getIdLong();
+        // do not need to test if key is present, shouldHandle(event) already does.
+        if (channelMonitor.isChannelBusy(id)) {
+            // todo check if /free called by original author, if not put message asking if he
+            // approves
+            channelMonitor.setChannelFree(id);
+            displayStatus(channelMonitor.getStatusChannelFor(event.getGuild()));
+            event.reply(UserStrings.MARK_AS_FREE.message()).queue();
+        } else {
+            Util.sendErrorMessage(event, UserStrings.ALREADY_FREE_ERROR.message());
+        }
+    }
+
+    /**
+     * Private method to test event to see if it should be processed.
```

Comment:
"private method" this is an obsolete comment and might also easily outdate. remove

Reply:
fixed